hpat/tests/test_series.py

```diff
@@ -1756,13 +1756,129 @@ def test_impl(S, param_skipna):
             result = hpat_func(S, param_skipna)
             self.assertEqual(result, result_ref)
 
-    def test_series_value_counts(self):
+    def test_series_value_counts_numbers(self):
         def test_impl(S):
             return S.value_counts()
+
+        data_to_test = [[1, 2, 3, 1, 1, 3],
+                        [1, 2, 3, np.nan, 1, 3, np.nan, np.inf],
+                        [0.1, 3., np.nan, 3., 0.1, 3., np.nan, np.inf]]
+
         hpat_func = hpat.jit(test_impl)
 
-        S = pd.Series(['AA', 'BB', 'C', 'AA', 'C', 'AA'])
-        pd.testing.assert_series_equal(hpat_func(S), test_impl(S))
+        for data in data_to_test:
+            S = pd.Series(data)
+            pd.testing.assert_series_equal(hpat_func(S), test_impl(S))
+
+    def test_series_value_counts_numbers_all_unique(self):
+        ''' Pandas and SDC results are slightly different because of sorting unique values '''
+        def test_impl(S):
+            return S.value_counts()
+
+        data_to_test = [[1, 2, 3, 4, 5],
+                        [5, 2, 1, 3, 4],
+                        [10, 9, 8, np.nan, np.inf],
+                        [0.1, 0.2, 0.3, 0.4, 0.5],
+                        [1., 2., 3., 4., 5., np.nan, np.inf]]
+
+        hpat_func = hpat.jit(test_impl)
+
+        for data in data_to_test:
+            S = pd.Series(data)
+            pd.testing.assert_series_equal(hpat_func(S).sort_index(), test_impl(S).sort_index())
+
+    def test_series_value_counts_sort(self):
```

Comment:
Tests `test_series_value_counts_sort` and `test_series_value_counts_sort_ascending` look similar. Let's merge them using loop on parameter `ascending`.